crates/sui-indexer-alt-graphql/src/api/types/transaction/filter.rs

```diff
@@ -24,6 +24,22 @@ pub(crate) struct TransactionFilter {
 
     /// Filter to transaction that occurred strictly before the given checkpoint.
     pub before_checkpoint: Option<UInt53>,
+
+    /// Limit to transactions that interacted with the given address.
+    /// The address could be a sender, sponsor, or recipient of the transaction.
+    pub affected_address: Option<SuiAddress>,
+}
+
+pub(crate) struct TransactionFilterValidator;
+
+impl CustomValidator<TransactionFilter> for TransactionFilterValidator {
+    fn check(&self, filter: &TransactionFilter) -> Result<(), InputValueError<TransactionFilter>> {
+        if filter.affected_address.is_some() {
+            // todo add validation
+        }
```

Comment:
What additional validation is required here? I think we get some validation from the `SuiAddress` input type.

Reply:
The way I would write this validation is something like:

```rust
let bounds = filter.affected_address.is_some() as u8
    + filter.affected_object.is_some() as u8
    + filter.function.is_some() as u8
    + filter.kind.is_some() as u8;

if bounds > 1 {
  return Err(...);
}
```

In case that impacts how you stub this function.